sssd_test_framework/hosts/client.py

```diff
@@ -30,6 +30,10 @@ def __init__(self, *args, **kwargs) -> None:
 
         self._features: dict[str, bool] | None = None
 
+    def pytest_setup(self) -> None:
+        result = self.conn.run("systemctl show sssd --value --property User")
+        self._sssd_service_user = result.stdout.strip()
```

Comment:
Set this property in constructor to default value and set real value here. Also remove the underscore. This is a public property since you are using it outside of this class.

I.e.

```python
def __init__(...):
    self.sssd_service_user: str = "root"
    """Default SSSD service user."""

def pytest_setup(self) -> None:
    super().pytest_setup()
    self.sssd_service_user = self.svc.get_property(...)
```

This way, it will be correctly documented.

Reply:
Fixed (but I don't it to defaults to 'root')

Reply:
Ok, then it can be done like this:

```
def __init__(self):
    self.__sssd_service_user: str | None = None

@property
def sssd_service_user(self):
    if self.__sssd_service_user is None:
        raise ValueError(...)

    return self.__sssd_service_user
```

But it's probably not necessary in this case.